iceoryx_posh/source/roudi/application/roudi_app.cpp

```diff
@@ -92,6 +92,11 @@ RouDiApp::RouDiApp(const config::CmdLineArgs_t& cmdLineArgs, const RouDiConfig_t
     {
         popo::internal::setUniqueRouDiId(cmdLineArgs.uniqueRouDiId.value());
     }
+    else
+    {
+        /// @todo iox-#861: we need a default RouDi ID. Which one to use
+        popo::internal::setUniqueRouDiId(1);
```

Comment:
I somehow feel this should not be necessary. Can we set the `UniqueRouDiId` to a default value and don't call the error handler if it is not set to get rid of all the `popo::internal::setUniqueRouDiId` in the tests?

Currently it runs with `0` when not explicitly set and works fine due to the mistake with the error handler. I do not see a reason why we need to enforce the setting when we have a working default value.

@elfenpiff what do you think?

Reply:
@elBoberido I completely agree. But the default value should be calculated as described in the comment of the single process example.

Reply:
@elfenpiff I would like to refactor the `TypedUniqueId` a bit. Instead of having a `setRouDiId` free function I would add a `TypedUniqueId::setDomainId`. This static function will call the error handler if there is already an instance of the `TypedUniqueId`. Therefore it must be called only once and before `TypedUniqueId` is used. For tests, the error handler can be overwritten. What do you think.

Reply:
@elfenpiff forget my suggestion. Since `TypedUniqueId` is a template, one cannot use my proposal. What do you think of the code I just committed?